src/mpyl/utilities/repo/__init__.py

```diff
@@ -73,14 +73,21 @@ def latest_tag(self) -> str:
     def changes_in_branch(self) -> list[Revision]:
         revisions = reversed(list(self._repo.iter_commits(f"{self._config.main_branch}..HEAD")))
         return [Revision(count, str(rev),
-                         self._repo.git.diff_tree(no_commit_id=True, name_only=True, r=str(rev)).splitlines()) for
-                count, rev in enumerate(revisions)]
+                         self._repo.git.diff_tree(no_commit_id=True, name_only=True, r=str(rev)).splitlines())
+                for count, rev in enumerate(revisions)]
 
     def changes_in_branch_including_local(self) -> list[Revision]:
         in_branch = self.changes_in_branch()
         in_branch.append(Revision(len(in_branch), self.get_sha, self.changes_in_commit()))
         return in_branch
 
+    def changes_in_merge_commit(self) -> list[Revision]:
+        curr_rev = self._repo.rev_parse('HEAD')
+        parent_revs = reversed(list(curr_rev.iter_parents(n=1)))
+        return [Revision(count, str(rev),
+                         self._repo.git.diff_tree(no_commit_id=True, name_only=True, r=str(rev)).splitlines())
+                for count, rev in enumerate(parent_revs)]
```

Comment:
We assume that `HEAD` points to a tagged revision. Perhaps you could add a verification to make sure that you're indeed looking at the expected tag?
In that case you could also throw a clear error if we're not looking at a merge commit.

Reply:
My current thinking is that I can get the tag associated with the commit by doing `git describe --tags curr_rev`. This should match the $TAG_NAME in jenkins that we pass to mpyl by `--tag $TAG_NAME`. So indeed what you suggested, the tag will be a parameter not a flag